infra/src/main/kotlin/fi/fta/geoviite/infra/util/Either.kt

```diff
@@ -0,0 +1,89 @@
+package fi.fta.geoviite.infra.util
+
+sealed class Either<out L, out R> {
```

Comment:
👍 Vaihtoehtona olisi varmaan ainakin Arrow-paketin käyttäminen halutuilta osilta (esim. Either https://arrow-kt.io/learn/typed-errors/either-and-ior/). Tuolla on myös jotain muitakin mielenkiintoisia tyyppejä, esim. nonEmptyList. Tosin ei ehkä kannata parin tyypin takia tuoda kokonaista dependencyä

(Arrow'n esimerkeissäkin muuten virheet vasemmalla)

Reply:
Arrowia voisi kyllä pöyhäistä vähän tarkemmin, sillä mielestäni siellä saattaisi olla muitakin kivoja funktionaalisia kavereita. Mutta pelkän Eitherin takia en itsekään sen mukaan ottamista kannata